src/core/ext/client_channel/client_channel.c

```diff
@@ -426,6 +433,18 @@ static void cc_start_transport_op(grpc_exec_ctx *exec_ctx,
   gpr_mu_unlock(&chand->mu);
 }
 
+static void cc_get_channel_info(grpc_exec_ctx *exec_ctx,
+                                grpc_channel_element *elem,
+                                grpc_channel_info *info) {
+  channel_data *chand = elem->channel_data;
+  gpr_mu_lock(&chand->mu);
+  if (info->lb_policy_name != NULL) {
+    *info->lb_policy_name = chand->lb_policy_name == NULL
+                                ? NULL : gpr_strdup(chand->lb_policy_name);
```

Comment:
The fact that the caller owns this memory is documented in the `grpc_channel_info` struct in grpc_types.h.

The intent of `grpc_channel_info` is to use the same sort of API as `grpc_op`: each field is a pointer to a caller-owned variable, which will be populated by the appropriate filter.  Because the caller owns the variables (and has explicitly asked the channel to populate them), it is responsible for freeing them.

I don't think it makes sense to have a `grpc_channel_info_destroy()`, any more than it would make sense to have a `grpc_op_destroy()`.  The caller already knows which field(s) were requested, so it can clean them up individually as appropriate.